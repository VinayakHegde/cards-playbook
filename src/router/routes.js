export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
