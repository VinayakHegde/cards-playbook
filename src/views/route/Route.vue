<style lang="scss" src="./style.scss"></style>

<script>
import Header from "@/components/header/Header.vue";
import Loader from "@/components/loader/Loader.vue";
export default {
  components: {
    Header,
    Loader
  }
};
</script>

<template>
  <div id="cards-playbook">
    <Loader v-if="$store.getters.loading" />
    <Header v-bind:text="$route.meta.header" />
    <router-view />
  </div>
</template>
