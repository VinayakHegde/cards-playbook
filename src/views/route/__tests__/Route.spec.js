import Vuex from "vuex";
import VueRouter from "vue-router";
import { mount, createLocalVue } from "@vue/test-utils";
import storeConfig from "@/store/store-config";
import routes from "@/router/routes";
import Route from "../Route.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
describe("Route.vue", () => {
  it("match the snapshot", () => {
    const router = new VueRouter({ routes });
    const store = new Vuex.Store(storeConfig);
    const wrapper = mount(Route, {
      localVue,
      router,
      store
    });
    expect(wrapper).toMatchSnapshot();
  });
});
