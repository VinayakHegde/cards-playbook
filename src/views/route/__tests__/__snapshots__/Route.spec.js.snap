// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route.vue match the snapshot 1`] = `
<div id="cards-playbook">
  <!---->
  <header></header>
  <!---->
</div>
`;
